src/components/FolderCard.jsx

```diff
@@ -13,38 +21,108 @@ const FolderCard = ({
 }) => {
+  const [isFolderModalOpen, setIsFolderModalOpen] = useState(false)
+  const [canShowDropdown, setCanShowDropdown] = useState(false)
+  const [canShowDeleteWarningModal, setCanShowDeleteWarningModal] = useState(false)
+  const dropdownRef = useRef(null)
+
+  useEffect(() => {
+    if (canShowDropdown) dropdownRef.current.focus()
+  }, [canShowDropdown])
+  
   const generateLink = () => {
     if (isHomepage) return `/sites/${siteName}/homepage`
     if (isCollection) return `/sites/${siteName}/collections/${category}`
     return `/sites/${siteName}/resources/${category}`
   }
 
-  return (
-    <Link className={`${contentStyles.component} ${contentStyles.card} ${elementStyles.folderCard}`} to={generateLink()}>
-      <div id={itemIndex} className={contentStyles.folderInfo}>
-        <i className={`bx bx-md text-dark ${isHomepage ? 'bxs-home-circle' : 'bxs-folder'}`} />
-        <span className={`${contentStyles.componentFolderName} align-self-center ml-4 mr-auto`}>{displayText}</span>
-        {
-          isHomepage
-          ? ''
-          : (
-            <div className={contentStyles.componentIcon}>
-              <button
-                className={contentStyles.componentIcon}
-                type="button"
-                id={`settings-folder-${itemIndex}`}
-                onClick={(e) => {
-                  e.preventDefault();
-                  settingsToggle(e)}}
-                className={contentStyles.componentIcon}
-              >
-                <i id={`settingsIcon-${itemIndex}`} className="bx bx-dots-vertical-rounded" />
-              </button>
-            </div>
-          )
-        }
+  const MenuItem = ({handler, id, children}) => {
+    return (
+      <div
+        id={id}
+        onClick={(e) => {
+          e.stopPropagation();
+          e.preventDefault(); 
+          if (handler) handler(e);
+        }}
+        className={`${elementStyles.dropdownItem}`}
+      >
+        {children}
       </div>
-    </Link>
+    )
+  }
+
+  const deleteHandler = async () => {
+    try {
+      const apiUrl = `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/${isCollection ? `/collections/${category}` : `/resources/${category}`}`
+      await axios.delete(apiUrl);
+
+      // Refresh page
+      window.location.reload();
+    } catch (err) {
+      console.log(err);
+    }
+  }
+
+  return (
+    <>
+      { isFolderModalOpen &&
+        <FolderModal
+          displayTitle={isCollection ? 'Rename Collection' : 'Rename Resource Category'}
+          displayText={isCollection ? 'Collection name' : "Resource category name"}
+          onClose={() => setIsFolderModalOpen(false)}
+          category={category}
+          siteName={siteName}
+          isCollection={isCollection}
+        />
+      }
+      { canShowDeleteWarningModal &&
+        <DeleteWarningModal
+          onCancel={() => setCanShowDeleteWarningModal(false)}
+          onDelete={deleteHandler}
+          type="folder"
+        />
```

Comment:
This should have been changed before, but can we also align the delete warning modal's buttons to the right?

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/245/commits/583bedc98151581efcf6afe1a21147cf9d1bbcc9